bindings/c/src/lib.rs

```diff
@@ -205,6 +206,187 @@ pub unsafe extern "C" fn libsql_disconnect(conn: libsql_connection_t) {
+        stmt.params.resize(idx, libsql::Value::Null);
+    }
+    stmt.params[idx - 1] = value.into();
+    0
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_bind_float(
+    stmt: libsql_stmt_t,
+    idx: std::ffi::c_int,
+    value: std::ffi::c_double,
+    out_err_msg: *mut *const std::ffi::c_char,
+) -> std::ffi::c_int {
+    let idx: usize = match idx.try_into() {
+        Ok(x) => x,
+        Err(e) => {
+            set_err_msg(format!("Wrong param index: {}", e), out_err_msg);
+            return 1;
+        }
+    };
+    let stmt = stmt.get_ref_mut();
+    if stmt.params.len() < idx {
+        stmt.params.resize(idx, libsql::Value::Null);
+    }
+    stmt.params[idx - 1] = value.into();
+    0
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_bind_null(
+    stmt: libsql_stmt_t,
+    idx: std::ffi::c_int,
+    out_err_msg: *mut *const std::ffi::c_char,
+) -> std::ffi::c_int {
+    let idx: usize = match idx.try_into() {
+        Ok(x) => x,
+        Err(e) => {
+            set_err_msg(format!("Wrong param index: {}", e), out_err_msg);
+            return 1;
+        }
+    };
+    let stmt = stmt.get_ref_mut();
+    if stmt.params.len() < idx {
+        stmt.params.resize(idx, libsql::Value::Null);
+    }
+    stmt.params[idx - 1] = libsql::Value::Null;
+    0
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_bind_string(
+    stmt: libsql_stmt_t,
+    idx: std::ffi::c_int,
+    value: *const std::ffi::c_char,
+    out_err_msg: *mut *const std::ffi::c_char,
+) -> std::ffi::c_int {
+    let idx: usize = match idx.try_into() {
+        Ok(x) => x,
+        Err(e) => {
+            set_err_msg(format!("Wrong param index: {}", e), out_err_msg);
+            return 1;
+        }
+    };
+    let value = unsafe { std::ffi::CStr::from_ptr(value) };
+    let value = match value.to_str() {
+        Ok(v) => v,
+        Err(e) => {
+            set_err_msg(format!("Wrong param value: {}", e), out_err_msg);
+            return 2;
+        }
+    };
+    let stmt = stmt.get_ref_mut();
+    if stmt.params.len() < idx {
+        stmt.params.resize(idx, libsql::Value::Null);
+    }
+    stmt.params[idx - 1] = value.to_string().into();
+    0
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_bind_blob(
+    stmt: libsql_stmt_t,
+    idx: std::ffi::c_int,
+    value: *const std::ffi::c_uchar,
+    value_len: std::ffi::c_int,
+    out_err_msg: *mut *const std::ffi::c_char,
+) -> std::ffi::c_int {
+    let idx: usize = match idx.try_into() {
+        Ok(x) => x,
+        Err(e) => {
+            set_err_msg(format!("Wrong param index: {}", e), out_err_msg);
+            return 1;
+        }
+    };
+    let value_len: usize = match value_len.try_into() {
+        Ok(v) => v,
+        Err(e) => {
+            set_err_msg(format!("Wrong param value len: {}", e), out_err_msg);
+            return 2;
+        }
+    };
+    let value = unsafe { core::slice::from_raw_parts(value, value_len) };
+    let value = Vec::from(value);
```

Comment:
This seems correct to me since its making a copy since we have no control over the allocator thus doing a Vec::from_raw_parts is not going to work.